yesod-form/Yesod/Form/Types.hs

```diff
@@ -45,6 +45,8 @@ import Data.Foldable
 --
 -- The 'Applicative' instance will concatenate the failure messages in two
 -- 'FormResult's.
+-- The 'Alternative' instance will choose 'FormFailure' before 'FormSuccess',
```

Comment:
@snoyberg: I think that might be from the cleanup :).

I've pushed a concise version that (I believe) has the correct behavior.